<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [6]:
# Membuat dataframes dari dictionary
import pandas as pd
df = pd.DataFrame({"Numbers":[1,2,3,4,5], "Letters":['a', 'b','c','d','e'], "Symbol":['!','@','#','$','%']})
df.head()

Numbers Letters Symbol
0        1       a      !
1        2       b      @
2        3       c      #
3        4       d      $
4        5       e      %

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [9]:
# Membaca data dan memasukannya kedalam suatu variabel
df=pd.read_csv('diamonds-1.csv')

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:
1.
List adalah kumpulan objek di python (tanpa library) 1 dimensi
Series adalah kumpulan objek di library pandas 1 dimensi dengan informasi index yang jelas dan bisa melakukan operasi vektor
Dictionary adalah pasangan key dan value, tidak memiliki indeks.

2.
Dataframe tidak memiliki indeks, pencarian data/akses data menggunakan key. Series lebih terstruktur dan berurutan karena memiliki indeks.

In [13]:
# code here

import pandas as pd

x = pd.Series([1,1,1,1,1,1])
x.head()

0    1
1    1
2    1
3    1
4    1
dtype: int64

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [22]:
# code here
df['price'].head()


0    326
1    326
2    327
3    334
4    335
Name: price, dtype: int64

In [21]:
df['color'].tail()

53935    D
53936    D
53937    D
53938    H
53939    D
Name: color, dtype: object

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: penjelasan mengenai statistika suatu data, seperti mean, median, mode, std, dan lain-lain

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [35]:
# code here
# df['price'].describe()
# df['depth'].describe()
df['carat'].describe()

count    53940.000000
mean         0.797940
std          0.474011
min          0.200000
25%          0.400000
50%          0.700000
75%          1.040000
max          5.010000
Name: carat, dtype: float64

Jawab:
df['price'].describe()
output: mean      3932.799722

df['depth'].describe()
output: std          1.432621

df['carat'].describe()
output: max          5.010000
